rest_framework/fields.py

```diff
@@ -59,7 +59,11 @@ def is_simple_callable(obj):
     if not (function or method):
         return False
 
-    args, _, _, defaults = inspect.getargspec(obj)
+    if six.PY2:
+        args, _, _, defaults = inspect.getargspec(obj)
+    else:
+        args, _, _, defaults = inspect.getfullargspec(obj)[:4]
+
```

Comment:
@tomchristie - I don't think a `compat.getargspec` makes sense here, as `signature` and `getargspec` behave differently and aren't directly interchangeable. Primarily, `signature` will not return `self` for bound methods where as `getargspec` will. I'll push a test case to demonstrate.